### PR TITLE
[PowerPC] Add test after #75271 on PPC. NFC.

### PR DESCRIPTION
**Pull request by bzEq**
Demonstrate `IMPLICIT_DEF implicit-def ...` can be generated after coalescing on PPC.

The case is reduced from failure in #75570. The failure is triggered after #75271 .

### CODE REVIEW

File: llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir [hunk 1]
```diff
@@ -0,0 +1,9039 @@
+# RUN: llc -mtriple=powerpc64le-unknown-linux-gnu -verify-coalescing -run-pass=register-coalescer \
+# RUN:   -o - %s | FileCheck %s
+--- |
```
[the remark this answers is not in the document]
Comment (bzEq) [reply llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir:3]: Tried `llvm-reduce`, still 5000+ lines of mir. I'll try to simplify LLVM IR...

[the remark this answers is not in the document]
Comment (bzEq) [reply llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir]: Thanks @arsenm , most of the 5000+ lines are virtregs, I've removed them.

File: llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir [hunk 2]
```diff
@@ -0,0 +1,84 @@
+# RUN: llc -mtriple=powerpc64le-unknown-linux-gnu -verify-coalescing -run-pass=register-coalescer \
+# RUN:   -o - %s | FileCheck %s
+---
+name:            _Z13testTransposeIfLj31ELj17EEvv
+alignment:       16
+exposesReturnsTwice: false
```
Comment (arsenm) [inline llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir:6]: Can drop most of these fields (running with -simplify-mir -run-pass=none will also do that)

File: llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+constants:       []
+machineFunctionInfo: {}
+body:             |
+  ; CHECK-LABEL: name:            _Z13testTransposeIfLj31ELj17EEvv
+  ; CHECK: undef %[[REG:[0-9]+]].sub_64:vsrc = IMPLICIT_DEF implicit-def %[[REG]]
+  bb.0:
+    successors: %bb.2(0x80000000)
+    liveins: $x2
+  
+    %5081:vssrc = IMPLICIT_DEF
```
Comment (arsenm) [inline llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir:70]: -run-pass=none will compact these numbers

File: llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+callsEHReturn:   false
+callsUnwindInit: false
+hasEHCatchret:   false
+hasEHScopes:     false
+hasEHFunclets:   false
+isOutlined:      false
+debugInstrRef:   false
+failsVerification: false
+tracksDebugUserValues: false
+registers:
```
Comment (arsenm) [inline llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir:22]: Can drop the section too, the uses have the inline class specifiers

**Review by arsenm — APPROVED** [review]

File: llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# RUN:   -o - %s | FileCheck %s
+---
+name:            _Z13testTransposeIfLj31ELj17EEvv
+alignment:       16
+tracksRegLiveness: true
+frameInfo:
+  maxAlignment:    128
+stack:
+  - { id: 0, size: 128, alignment: 128, local-offset: -128 }
+  - { id: 1, size: 4096, alignment: 16, local-offset: -4224 }
```
Comment (nemanjai) [inline llvm/test/CodeGen/PowerPC/coalescer-remat-with-undef-implicit-def-operand.mir:11]: Nit (feel free to ignore): there don't seem to be any stack accesses here. Do we need these?

**Review by nemanjai — APPROVED** [review]
LGTM.

**Review by arsenm — APPROVED** [review]

**Comment by aeubanks** [comment]
sorry, I had to revert this since in ece1359857c547a156ed743643bccbfd0f09bf2a since it depended on another commit reverted in c4146121e940b6b853148c780568dee38b97382f

**Pull request merged**